### PR TITLE
Update to Kotlin 1.8.0

### PR DESCRIPTION
**Pull request by ZacSweers**
This updates to Kotlin 1.8.0 for both the repo _and_ its configuration logic to use new KGP 1.8.0 APIs (namely `compilerOptions`). With this, this plugin now requires kotlin 1.8.0 or later.

### CODE REVIEW

**Comment by ZacSweers** [comment]
I've released `0.3.6-1-8-0-RC` with the changes up to this point

**Comment by ZacSweers** [comment]
And now `0.3.7-1-8-0-RC` with the latest changes from 0.3.7

**Comment by ZacSweers** [comment]
And now 0.3.8-1-8-0-RC2 with the latest changes from 0.3.8 + kotlin 1.8.0-RC2

File: slack-plugin/build.gradle.kts [hunk 1]
```diff
@@ -5,6 +5,10 @@ plugins {
   alias(libs.plugins.bestPracticesPlugin)
 }
 
+kotlinDslPluginOptions {
+  jvmTarget.set("11")
```
Comment (ZacSweers) [inline slack-plugin/build.gradle.kts:9]: Found out the hard way that kotlin-dsl overrides your jvm target too unless you set this 🙃

**Comment by ZacSweers** [comment]
Going to get a retroactive review on this one given the small set of changes, prior integration test in the slack android repo, and desire to move this on before I go back on vacation

**Review by autonomousapps — COMMENTED** [review]
LGTM.

**Pull request merged**